src/main/java/com/yevhenberladyniuk/debts/service/PartnerServiceImpl.java

```diff
@@ -22,42 +23,52 @@ public PartnerServiceImpl(PartnerRepository partnerRepository) {
     }
 
     @Override
-    public void create(CreatePartnerForm createPartnerForm, Long userId) {
+    public void create(CreatePartnerForm createPartnerForm, User user) {
 
         Partner partner = Partner.builder()
-                .userId(userId)
+                .userId(user.getId())
                 .firstName(createPartnerForm.getFirstName())
                 .lastName(createPartnerForm.getLastName())
                 .debt(createPartnerForm.getDebt())
-                .date(LocalDateTime.now())
+                .updatedAt(LocalDateTime.now())
+                .active(true)
                 .build();
 
         partnerRepository.save(partner);
     }
 
     @Override
-    public List<Partner> findAllByUserId(Long userId) {
-        return partnerRepository.findAllByUserId(userId);
+    public List<Partner> findAll(User user) {
+        return partnerRepository.findAllByUserIdAndActive(user.getId(), true);
     }
 
     @Override
-    public void deletePartnerById(Long id) {
-        partnerRepository.deleteById(id);
+    public void deletePartnerById(Long id, User user) {
+        Partner partner = findPartnerById(id, user);
+        partner.setActive(false);
+        partnerRepository.save(partner);
     }
 
     @Override
-    public Optional<Partner> findPartnerById(Long id) {
-        return partnerRepository.findById(id);
+    public Partner findPartnerById(Long id, User user) {
+
+        Optional<Partner> optionalPartner = partnerRepository.findByIdAndActive(id, true);
+        Partner partner = optionalPartner.orElseThrow( () -> new RuntimeException("Partner not found"));
+
+        if(partner.getUserId().equals(user.getId())){
+            return partner;
+        }
+
+        throw new RuntimeException("Partner not found");
```

Comment:
When you add `findByIdAndActiveAndUsedId` method, no need if this userId validation